pkg/controller/utils/ca.go

```diff
@@ -0,0 +1,460 @@
+package utils
```

Comment:
What was the reasoning behind putting the implementations of the interfaces in the tls package here? I would think this is something in the tls package as the default implementation of those interfaces (either at the same package level of the interfaces or as a subpackage). At the very least I don't think it belongs alongside all the other utilities here (this utils folder in general probably shouldn't exist, it's bad practice as it defers properly organising the code into relavant packages).